openapi/v2/core-metadata.yaml

```diff
@@ -3108,6 +3287,496 @@ paths:
+        required: true
+        schema:
+          type: string
+          format: uuid
+        description: "An ID of datatype string, by default a GUID."
+    get:
+      summary: "Returns a provision watcher by ID"
+      responses:
+        '200':
+          description: "OK"
+          headers:
+            X-Correlation-ID:
+              $ref: '#/components/headers/correlatedResponseHeader'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ProvisionWatcherResponse'
+              example:
+                apiVersion: "v2"
+                requestId: "4e63c9b9-9cc4-45e4-9bf0-6d994a3860b4"
+                statusCode: 200
+                message: ""
+                provisionwatcher:
+                  apiVersion: "v2"
+                  id: "1ff7762f-c432-4af0-9a5d-756bbc92744b"
+                  name: "virtual-watcher"
+                  created: 1600927134890
+                  modified: 1600927134890
+                  labels:
+                    - virtual
+                  identifiers:
+                    "address": "localhost"
+                    "port": "3[0-9]{2}"
+                  blockingIdentifiers:
+                    "port":
+                      - "397"
+                      - "398"
+                      - "399"
+                  profile: "device-virtual"
+                  service: "device-virtual"
+                  adminState: "UNLOCKED"
+                  autoEvents:
+                    - resource: "Bool"
+                      frequency: "10s"
+                      onChange: false
+        '400':
+          description: "Request is in an invalid state"
+          headers:
+            X-Correlation-ID:
+              $ref: '#/components/headers/correlatedResponseHeader'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              examples:
+                400Example:
+                  $ref: '#/components/examples/400Example'
+        '404':
+          description: "The requested resource does not exist"
+          headers:
+            X-Correlation-ID:
+              $ref: '#/components/headers/correlatedResponseHeader'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              examples:
+                404Example:
+                  $ref: '#/components/examples/404Example'
+        '500':
+          description: "Internal Server Error"
+          headers:
+            X-Correlation-ID:
+              $ref: '#/components/headers/correlatedResponseHeader'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              examples:
+                500Example:
+                  $ref: '#/components/examples/500Example'
+    delete:
+      summary: "Delete a provision watcher by ID"
```

Comment:
@cloudxxx8 , since we are not doing by Id for queries doesn't it make since to be consistent and not have by Id for deletes? 

Not sure why I did catch this before now... :-(

Reply:
maybe bring up this topic in this week's community meeting ?

Reply:
@lenny-intel we talked about maybe we will add query by id back if lots of users say they need it.
for delete by id, we have implemented for all APIs.  the reason I keep it is that users might want to query multiple instances by certain conditions, and they will have ids at that point.  Then, deleting lots of instances by ids has better performance than deleting by names.

We can discuss in the WG meeting and make the decision.